client/src/dojo/modelManager/ConfigManager.ts

```
@@ -478,10 +473,70 @@ export class ClientConfigManager {
-      return total + this.getResourceRarity(resource) * amount;
-    }, 0);
+  getHyperstructureTotalContributableAmount(hyperstructureId: number) {
+    const hyperstructure = getComponentValue(
+      this.components.Hyperstructure,
+      getEntityIdFromKeys([BigInt(hyperstructureId)]),
+    );
+
+    let total = 0;
+    const randomness = Number(hyperstructure?.randomness ?? 0);
+    for (const tier in ResourceTier) {
+      if (isNaN(Number(tier))) continue; // Skip non-numeric enum values
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)